### PR TITLE
Add Spark extension

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

Add Spark extension

### CODE REVIEW

File: extensions/spark/src/main/scala/ai/djl/spark/SparkModel.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import ai.djl.modality.Classifications
+import ai.djl.repository.zoo.{Criteria, ZooModel}
+import ai.djl.training.util.ProgressBar
+import ai.djl.translate.Translator
+import org.apache.spark.sql.Row
+
+import java.io.Serializable
+
+/**
+ * `SparkModel` is the implementation of [[ai.djl.Model]] for Spark support.
```
Comment (frankfliu) [inline extensions/spark/src/main/scala/ai/djl/spark/SparkModel.scala:25]: does scala doc support markdown?

File: extensions/spark/build.sbt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name := "spark"
```
Comment (frankfliu) [inline extensions/spark/build.sbt:1]: Any reason we add build.sbt?

File: extensions/spark/src/main/scala/ai/djl/spark/translator/SparkImageClassificationTranslator.scala [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import ai.djl.ndarray.types.{DataType, Shape}
+import ai.djl.translate.{Batchifier, Pipeline, Translator, TranslatorContext}
+import ai.djl.util.Utils
+import org.apache.spark.ml.image.ImageSchema
+import org.apache.spark.sql.Row
+
+import java.util
+
+/** A [[ai.djl.translate.Translator]] for Spark Image Classification tasks. */
+@SerialVersionUID(123456789L)
```
Comment (frankfliu) [inline extensions/spark/src/main/scala/ai/djl/spark/translator/SparkImageClassificationTranslator.scala:27]:
```suggestion
@SerialVersionUID(1L)
```

File: extensions/spark/build.sbt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+version := "0.1"
+
+scalaVersion := "2.12.10"
+scalacOptions += "-target:jvm-1.8"
+
+resolvers += Resolver.jcenterRepo
+
+libraryDependencies += "org.apache.spark" %% "spark-core" % "3.0.1"
```
Comment (lanking520) [inline extensions/spark/build.sbt:10]: any reason we are using 3.0.1? Do we also want to put this into gradle.properties?

**Review by lanking520 — COMMENTED** [review]
Please consolidate versioning in a doc. By the way, could you also create a simple Python wrapper to demo the capability in PySpark?

File: extensions/spark/src/main/scala/ai/djl/spark/SparkModel.scala (hunk 1 above)
Comment (xyang16) [reply extensions/spark/src/main/scala/ai/djl/spark/SparkModel.scala:25]: Yes. This is the scala format of scaladoc

File: extensions/spark/build.sbt (hunk 2 above)
Comment (xyang16) [reply extensions/spark/build.sbt]: Removed.

File: extensions/spark/src/main/scala/ai/djl/spark/translator/SparkImageClassificationTranslator.scala (hunk 3 above)
Comment (xyang16) [reply extensions/spark/src/main/scala/ai/djl/spark/translator/SparkImageClassificationTranslator.scala]: Changed.

File: extensions/spark/build.sbt (hunk 4 above)
Comment (xyang16) [reply extensions/spark/build.sbt]: OK, put into gradle.properties.

File: extensions/spark/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+## Installation
+
+You can pull the module from the central Maven repository by including the following dependency in your `pom.xml` file:
+
+```xml
+<dependency>
+    <groupId>ai.djl.spark</groupId>
+    <artifactId>spark</artifactId>
+    <version>0.19.0</version>
```
Comment (lanking520) [inline extensions/spark/README.md:23]: 0.20.0

**Review by lanking520 — APPROVED** [review]

**Pull request merged**